airbyte-webapp/src/views/Connector/ConnectorCard/ConnectorCard.tsx

```diff
@@ -227,33 +227,35 @@ export const ConnectorCard: React.FC<ConnectorCardCreateProps | ConnectorCardEdi
       onSubmit={onHandleSubmit}
       formValues={formValues}
       connectorId={isEditMode ? getConnectorId(props.connector) : undefined}
-      renderFooter={({ dirty, isSubmitting, isValid, resetConnectorForm, getValues }) => (
-        <Controls
-          isEditMode={Boolean(isEditMode)}
-          isTestConnectionInProgress={isTestConnectionInProgress}
-          onCancelTesting={onStopTesting}
-          isSubmitting={isSubmitting || isTestConnectionInProgress}
-          errorMessage={error && generateMessageFromError(error)}
-          formType={props.formType}
-          hasDefinition={Boolean(selectedConnectorDefinitionId)}
-          onRetestClick={() => {
-            if (!selectedConnectorDefinitionId) {
-              return;
-            }
-            handleTestConnector(
-              isEditMode ? undefined : { ...getValues(), serviceType: selectedConnectorDefinitionId }
-            );
-          }}
-          onDeleteClick={onDeleteClick}
-          isValid={isValid}
-          dirty={dirty}
-          job={job ? job : undefined}
-          onCancelClick={() => {
-            resetConnectorForm();
-          }}
-          connectionTestSuccess={connectionTestSuccess}
-        />
-      )}
+      renderFooter={({ dirty, isSubmitting, isValid, resetConnectorForm, getValues }) =>
+        selectedConnectorDefinitionSpecification && (
```

Comment:
Only render the controls if the user has selected a connector type, since it is weird to show the `Set up connector` button if nothing has been selected yet.